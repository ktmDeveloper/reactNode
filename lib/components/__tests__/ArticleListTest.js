import React from 'react';
import ArticleList from '../ArticleList';
import renderer from 'react-test-renderer';

describe('ArticleList', () => {
    
    const testProps = {
        articles: {
            a: {id:'a'},
            b: {id:'b'}
        },
        articleActions: {
            lookupAuthor: jest.fn(() => ({})) //jest.fn returns a fake function and returns a object
        }
    };
    it('renders correctly', () => {
        const tree = renderer.create(
            <ArticleList 
            {...testProps}/>
        ).toJSON();
        
        expect(tree.children.length).toBe(2);
        expect(tree).toMatchSnapshot(); //https://facebook.github.io/jest/docs/en/snapshot-testing.html
    });
});

