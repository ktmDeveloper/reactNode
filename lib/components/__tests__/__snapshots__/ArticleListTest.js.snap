// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleList renders correctly 1`] = `
<div>
  <div
    className="card"
  >
    <div
      className="article card-block"
    >
      <h1
        className="card-title"
      />
      <span
        className="card-subtitle mb-2 text-muted"
      >
        Invalid Date
      </span>
      <p
        className="card-test"
      />
      <a
        className="card-link"
        href={undefined}
      >
         
      </a>
    </div>
  </div>
  <div
    className="card"
  >
    <div
      className="article card-block"
    >
      <h1
        className="card-title"
      />
      <span
        className="card-subtitle mb-2 text-muted"
      >
        Invalid Date
      </span>
      <p
        className="card-test"
      />
      <a
        className="card-link"
        href={undefined}
      >
         
      </a>
    </div>
  </div>
</div>
`;
